From c3c28ed00601a2fbab8efbdc787cb0e1696b1fea Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 27 Jan 2026 17:48:43 +0100
Subject: [PATCH] Define `SIO_UDP_NETRESET` if necessary

Older `mingw-w64-headers` versions seem not to define this constant, but
OpenSSL started using it in v3.5.5.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 ssl/quic/quic_reactor.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ssl/quic/quic_reactor.c b/ssl/quic/quic_reactor.c
index 1a95f13..fd913c7 100644
--- a/ssl/quic/quic_reactor.c
+++ b/ssl/quic/quic_reactor.c
@@ -15,6 +15,9 @@
 #include <winsock2.h>
 #include <mstcpip.h>
 #include <mswsock.h>
+#if defined(__MINGW32__) && !defined(SIO_UDP_NETRESET)
+#define SIO_UDP_NETRESET _WSAIOW(IOC_VENDOR, 15)
+#endif
 #endif
 
 /*
-- 
2.52.0.windows.1

